tools/deviceGen/gen.py

```diff
@@ -0,0 +1,73 @@
+from DeviceGen import *
+from argparse import ArgumentParser
+from sys import exit
+
+if __name__ == "__main__":
+    parser = ArgumentParser(description="Generate Header Files")
+
+    parser.add_argument("limits_json_path", type=str, help="The path to the device_limits.json file")
```

Comment:
oh so you have some arguments, why they are not present in CMake invocation [here](https://github.com/Devsh-Graphics-Programming/Nabla/pull/643/files#r1641304386)?